guides/advanced_guides/faceted_search.md

```diff
@@ -0,0 +1,192 @@
+      "release_date": 1553644800
+  },
+  {
+      "id": "458723",
+      "title": "Us",
+      "director": "Jordan Peele",
+      "genre": "Horror",
+      "poster": "https://image.tmdb.org/t/p/w1280/ux2dU1jQ2ACIMShzB3yP93Udpzc.jpg",
+      "overview": "Husband and wife Gabe and Adelaide Wilson take their...",
+      "release_date": 1552521600
+  },
+  ...
+]
+```
+
+You would declare faceted attributes as follows:
+
+```bash
+$ curl \
+  -X POST 'http://localhost:7700/indexes/movies/settings' \
+  --data '{
+      "attributesForFaceting": [
+          "director",
+          "genre"
+      ]
+  }'
+```
+
+## Querying On Faceted Attributes
+
+When performing a search, you can pass parameter to your query:
+
+### 1. The facets to filter on
+
+You can filter on facets to narrow down your results based on criteria.
+
+`facetFilters=["facetName:facetValue"]`, `facetFilters=[["facetName:facetValue"]]` or a mix of both `facetFilters=["facetName1:facetValue1", ["facetName2:facetValue2"]]`
+
+- `["facetName1:facetValue1", ["facetName2:facetValue2"]]` (Array of array of strings or single strings, defaults to `null`)
+
+  Both types of array contain the facet names and values to filter on.
+  A valid array must be an array which contains either a list of strings or arrays of strings and can mix both (e.g. `["kind:t-shirt", ["color:red", "color:green"]]`).
+
+  - `facetName`: The name (the attribute) of a field used as a facet (e.g. `color`, `kind`).
+  - `facetValue`: The value of this facet to filter results on (e.g. `red`, `green`, `t-shirt`, `pants`).
+
+### Example
+
+Given a clothing dataset, suppose you want to retrieve all t-shirts with a nautical pattern. You would then use:
+
+```bash
+$ curl --get 'http://localhost:7700/indexes/clothing/search' \
+    --data-urlencode 'q=nautical' \
+    --data-urlencode 'facetFilters=["kind:t-shirt"]'
+```
+
+### Logical Connectives
+
+Inputting a double dimensional array allows you to use **logical connectives**.
+
+- **Inner arrays elements** are connected by an `OR` operator (e.g. `[["color:red", "color:green"]]`).
+- **Outer arrays elements** are connected by an `AND` operator (e.g. `["color:red", "kind:t-shirt"]`).
+
+You can mix connectives, for instance, the following array:
+
+```json
+["kind:t-shirt", ["color:red", "color:green"]]
+```
+
+Can be translated as:
+
+```SQL
+"kind:t-shirt" AND ("color:red" OR "color:green")
+```
+
+### Example
+
+Say you want to get movies matching "thriller" classified as either comedy or horror and directed by Jordan Peele.
+
+```SQL
+("genre:Horror" OR "genre:Comedy") AND "director:Jordan Peele"
+```
+
+Querying on "thriller", the above example results in the following CURL command:
+
+```bash
+$ curl --get 'http://localhost:7700/indexes/movies/search' \
+    --data-urlencode 'q=thriller' \
+    --data-urlencode 'facetFilters=[["genre:Horror", "genre:Comedy"], "director:Jordan Peele"]'
+```
+
+### 2. The facets for which to retrieve the matching count
```

Comment:
This section is not clear and I don't think it means what it should mean. 

```suggestion
### 2. The facets distribution
```
The `facets` parameter returns an object containing the number of matching documents distributed amongst all the values of a given facet. 

For example, when searching for `Batman`, the `facets` on `genre` may look like this: 
```json
{
  "facets": {
    "genre": {
      "thriller": 158,
      "comedy": 5,
      "romantic": 0
    }
  }
}
```
This mean that 158 documents containing the query `Batman` has the `thriller` genre, 5 have the `comedy` genre. While zero romantic movies have been found containing that same query. If the user is searching for a `comedy` movie around batman, he would use the `facetFilter` to narrow its search by only allowing the `comedy` genre and only the 5 matching document would be returned.

> The user uses this information to further its search in one or the other direction by using the `facetFilters`


### Use Cases

<img width="309" alt="Screenshot 2020-05-25 at 11 13 58" src="https://user-images.githubusercontent.com/33010418/82799411-4cce2b00-9e7a-11ea-9e32-624c247f2c47.png">

> Example when looking for categories on the [snowleader.co.uk website](https://www.snowleader.co.uk/en/catalogsearch/result/?universes-selector=all&q=boot)

In the example above, when searching for `Boots`, the `facets` inform us that 791 `outdoor` boots have been found in the `categories` facets. 

Another example on the `size` facet.
<img width="330" alt="Screenshot 2020-05-25 at 14 21 46" src="https://user-images.githubusercontent.com/33010418/82812340-1cdf5180-9e93-11ea-9945-792608d2cc67.png">
The user can narrow down its selection to all the pants available in `XXL` size. Thus not wasting time looking at pants not available in its size.

### Usage Examples

The `return` object si important in this selection. I will complete this section once I have the appropriate dataset